platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/EventCreationManagerWiring.java

```diff
@@ -0,0 +1,150 @@
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Instant;
+
+/**
+ * Wiring for the {@link EventCreationManager}.
+ */
+public class EventCreationManagerWiring {
+
+    private final TaskScheduler<GossipEvent> taskScheduler;
+
+    private final BindableInputWire<GossipEvent, GossipEvent> eventInput;
+    private final BindableInputWire<Long, GossipEvent> minimumGenerationNonAncientInput;
+    private final BindableInputWire<Boolean, GossipEvent> pauseInput;
+    private final Bindable<Instant, GossipEvent> heartbeatBindable;
+    private final OutputWire<GossipEvent> newEventOutput;
+
+    /**
+     * Create a new instance of this wiring.
+     *
+     * @param platformContext the platform context
+     * @param taskScheduler   the task scheduler for the event creation manager
+     * @return the new wiring instance
+     */
+    @NonNull
+    public static EventCreationManagerWiring create(
+            @NonNull final PlatformContext platformContext, @NonNull final TaskScheduler<GossipEvent> taskScheduler) {
+        return new EventCreationManagerWiring(platformContext, taskScheduler);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param platformContext the platform context
+     * @param taskScheduler   the task scheduler for the event creation manager
+     */
+    private EventCreationManagerWiring(
+            @NonNull final PlatformContext platformContext, @NonNull final TaskScheduler<GossipEvent> taskScheduler) {
+
+        this.taskScheduler = taskScheduler;
+
+        eventInput = taskScheduler.buildInputWire("possible parents");
+        minimumGenerationNonAncientInput = taskScheduler.buildInputWire("minimum generation non ancient");
+        pauseInput = taskScheduler.buildInputWire("pause");
+        newEventOutput = taskScheduler.getOutputWire();
+
+        final double frequency = platformContext
+                .getConfiguration()
+                .getConfigData(EventCreationConfig.class)
+                .creationAttemptRate();
+        heartbeatBindable = taskScheduler.buildHeartbeatInputWire("heartbeat", frequency);
+    }
+
+    /**
+     * Bind an event creation manager to this wiring.
+     *
+     * @param eventCreationManager the event creation manager to bind
+     */
+    public void bind(@NonNull final EventCreationManager eventCreationManager) {
+        eventInput.bind((@NonNull final GossipEvent event) -> {
+            // FUTURE WORK: once the feature flag has been removed,
+            // convert the internals of event creation to use GossipEvent
+            // instead of EventImpl.
+            final EventImpl eventImpl = new EventImpl(event.getHashedData(), event.getUnhashedData());
+            eventCreationManager.registerEvent(eventImpl);
+        });
+
+        minimumGenerationNonAncientInput.bind(eventCreationManager::setMinimumGenerationNonAncient);
+
+        pauseInput.bind(eventCreationManager::setPauseStatus);
+
+        heartbeatBindable.bind(now -> {
```

Comment:
I am assuming that this particular binding is responsible for triggering event creation.  How do I track down the source of the heartbeat?  Where is the frequency configured?  Formally documented?

Reply:
That I can look at a chunk of code and be inspired to ask this question is great! :-)

Reply:
The heartbeat frequency is configured in the constructor of this class:

```
        final double frequency = platformContext
                .getConfiguration()
                .getConfigData(EventCreationConfig.class)
                .creationAttemptRate();
        heartbeatBindable = taskScheduler.buildHeartbeatInputWire("heartbeat", frequency);
```

Heartbeat wires are a native feature of the wiring framework. You ask for an input wire with a specified frequency or period, and the framework will send it a message with the current time at the requested rate. In the diagram I link to in the PR description, you can see the `heartbeat` box that is automatically constructed by the framework.